components/dashboard/src/data/setup.tsx

```diff
@@ -78,26 +78,35 @@ function createIDBPersister(idbValidKey: IDBValidKey = "gitpodQueryClient"): Per
     // If we get an error performing an operation, we'll disable persistance and assume it's not supported
     let persistanceActive = true;
 
-    const throttle = pThrottle({
-        interval: 500,
-        limit: 1,
-        strict: true,
-    });
-
-    const throttledSet = throttle(async (client: PersistedClient) => {
-        await set(idbValidKey, client);
-    });
+    // Ensure we don't persist the client too frequently
+    // Important to debounce (not throttle) this so we aren't queuing up a bunch of writes
+    // but instead, only persist the latest state
+    const debouncedSet = debounce(
+        async (client: PersistedClient) => {
+            await set(idbValidKey, client);
+        },
+        500,
+        {
+            leading: true,
+            // important so we always persist the latest state when debouncing calls
+            trailing: true,
+            // ensure
+            maxWait: 1000,
```

Comment:
I'm trying to understand the change with cache persistence here. I guess you right, throttling with an interval of 500ms can still cause issues with queuing a lot of requests. The original source of the throttle recommendation didn't include proper settings, unfortunately.

Reading the change to use `debounce` with the `maxWait` option let me think that it would also be possible to achieve the very same solution by bumping the `interval` setting to 1000ms. Why? Because throttle is no more than a wrapper around `debounce` with `maxWait` ;-) Not sure? See it's [implementation](https://github.com/lodash/lodash/blob/4.17.11/lodash.js#L10898-L10914).

That said, it's awesome to learn the effect of such settings and also to get rid of 1/2 of the dependencies.

Reply:
After looking at the context of issue reported and to be fixed here, I'm pretty sure it the same problem that was tackled in https://github.com/gitpod-io/gitpod/pull/18082

The difference is, in this case we're facing a lot of data to be persisted, which produces the same queuing and business with the browser. Back then if was not necessary to have big chucks to be persisted, it was already burning when the simple stuff was cached.

Nice catch @selfcontained! 🎉